modules/mesh-task/templates/consul_client_command.tpl

```diff
@@ -3,7 +3,32 @@ cp /bin/consul /bin/consul-inject/consul
 ECS_IPV4=$(curl -s $ECS_CONTAINER_METADATA_URI_V4 | jq -r '.Networks[0].IPv4Addresses[0]')
 
 %{ if tls ~}
-echo "$CONSUL_CACERT" > /tmp/consul-ca-cert.pem
+echo "$CONSUL_CACERT" > /consul/consul-ca-cert.pem
+%{ endif ~}
+
+%{ if acls && client_token_auth_method_name != "" ~}
+
+login() {
+    echo "Logging into auth method: name=${ client_token_auth_method_name }"
+    consul login \
+      -http-addr ${ consul_http_addr } \
+    %{ if tls ~}
+      -ca-file /consul/consul-ca-cert.pem \
+    %{ endif ~}
+    %{ if consul_partition != "" ~}
+      -partition ${ consul_partition } \
+    %{ endif ~}
+      -type aws -method ${ client_token_auth_method_name } \
+      -aws-auto-bearer-token -aws-include-entity \
+      -token-sink-file /consul/token.txt
+}
+
+while ! login; do
+    sleep 2
+done
+
+# This is an env var which is interpolated into the agent-defaults.hcl
+export AGENT_TOKEN=$(cat /consul/token.txt)
```

Comment:
This env var will be set once after the login script completes successfully.. The docs for the [`-token-sink-file`](https://www.consul.io/commands/login#token-sink-file) option seem to suggest that the token secret ID in this file may be updated. I'm not sure under what circumstances this would happen, or if it even would.

I think what is here is fine, I had a look through the code and it appears that the file is only ever written to once, after a successful login.. unless I'm missing something: https://github.com/hashicorp/consul/blob/main/command/login/login.go#L159

Curious about your thoughts or if you checked into this.

Reply:
Right, the Consul client/agent doesn't rewrite that token file. Only the `consul login` command would update that file, if it already exists, in order to make it easy to refresh tokens. This would matter if we want to support token expiration (`consul acl auth-method create -max-token-ttl <duration> ...`) but we aren't supporting token expiration for now.